### PR TITLE
WG-Easy: Fix Add Variable for Listen Interface/Device

### PR DESCRIPTION
**Pull request by TCB13**
This is an initial PR to fix https://github.com/truenas/charts/issues/1369. It adds a variable on the interface so the user can choose in what interface WG should listen.

It would be very nice if someone with more experience in this project could "chip in" and help make this fix a reality. I'm not sure how to actually fix things futher than this.

Thank you.

### CODE REVIEW

**Review by stavros-k — CHANGES_REQUESTED** [review]
Hello, please make these changes in `library/ix-dev/charts/wg-easy` directory. 
Thanks

**Comment by TCB13** [comment]
@stavros-k all done :)

**Comment by TCB13** [comment]
> This only affects when hostNetwork is enabled, right? Otherwise it would use the interface from the container. (Which is `eth0`).
> 
> I think we should default this to `eth0` (As the upstream has it). And probably do not set the env var at all when hostNetwork is off, because changing it in this case it will "break" it


As you can see at https://github.com/wg-easy/wg-easy/blob/master/src/config.js#L9 , https://github.com/wg-easy/wg-easy/blob/master/src/config.js#L22 and at https://github.com/wg-easy/wg-easy/blob/master/README.md?plain=1#L86 the upstream replaced the static `eth0` with the `WG_DEVICE` env variable. It also fallbacks to `eth0` so nothing will break. In my commit I also set `eth0` as default value for the variable.

Either way, the current solution at truenas is broken with hostNetwork is enabled and disabling results in additional and unnecessary CPU load.

**Comment by stavros-k** [comment]
> > This only affects when hostNetwork is enabled, right? Otherwise it would use the interface from the container. (Which is `eth0`).
> > I think we should default this to `eth0` (As the upstream has it). And probably do not set the env var at all when hostNetwork is off, because changing it in this case it will "break" it
> 
> As you can see at https://github.com/wg-easy/wg-easy/blob/master/src/config.js#L9 , https://github.com/wg-easy/wg-easy/blob/master/src/config.js#L22 and at https://github.com/wg-easy/wg-easy/blob/master/README.md?plain=1#L86 the upstream replaced the static `eth0` with the `WG_DEVICE` env variable. It also fallbacks to `eth0` so nothing will break. In my commit I also set `eth0` as default value for the variable.
> 
> Either way, the current solution at truenas is broken with hostNetwork is enabled and disabling results in additional and unnecessary CPU load.

Okay, but we'll have to wait until a new container is release with this change first.

**Comment by TCB13** [comment]
Lets leave this open then. :)

**Comment by stavros-k** [comment]
Considering the upcoming migration #1403, I'll close this one. This is because it will end up with a lot of conflicts anyway.

Feel free to open a new PR after the linked migration is merged.
Otherwise I'll handle it once upstream releases a new image

Thanks

**Comment by TCB13** [comment]
> Feel free to open a new PR after the linked migration is merged.
> Otherwise I'll handle it once upstream releases a new image

I ended up abandoning TrueNAS altogether as most images on the "charts" are broken in one or more ways and the base system (TrueNAS Scale) doesn't seem mature enough for any kind of production usage.

Thank you anyways.

**Pull request closed without merging**
